src/components/stories/story-view/components/index.tsx

```diff
@@ -0,0 +1,206 @@
+import React, {
+  forwardRef,
+  memo,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+
+import {Image} from 'react-native';
+import {useSharedValue} from 'react-native-reanimated';
+
+import {StoryContainer} from './story-container';
+
+import {
+  ANIMATION_DURATION,
+  BACKGROUND_COLOR,
+  CLOSE_COLOR,
+  STORY_AVATAR_SIZE,
+} from '../core/constants';
+import {StoryContainerPublicMethods} from '../core/dto/componentsDTO';
+import {ProgressStorageProps} from '../core/dto/helpersDTO';
+import {
+  StoriesViewProps,
+  StoriesViewPublicMethods,
+} from '../core/dto/storiesViewDTO';
+import {
+  clearProgressStorage,
+  getProgressStorage,
+  setProgressStorage,
+} from '../core/helpers/storage';
+
+const StoriesView = forwardRef<StoriesViewPublicMethods, StoriesViewProps>(
+  (
+    {
+      stories,
+      saveProgress = false,
+      storyAvatarSize = STORY_AVATAR_SIZE,
+      animationDuration = ANIMATION_DURATION,
+      backgroundColor = BACKGROUND_COLOR,
+      videoAnimationMaxDuration,
+      videoProps,
+      closeIconColor = CLOSE_COLOR,
+      initialID,
+      ...props
+    },
+    ref,
+  ) => {
+    const [data, setData] = useState(stories);
+
+    const seenStories = useSharedValue<ProgressStorageProps>({});
+    const loadedStories = useSharedValue(false);
+    const loadingStory = useSharedValue<string | undefined>(undefined);
+
+    const modalRef = useRef<StoryContainerPublicMethods>(null);
+
+    useEffect(() => {
+      onPress(initialID);
+    }, [initialID]);
+
+    const onPress = (id: string) => {
+      loadingStory.value = id;
+      modalRef.current?.show(id);
+    };
+
+    const onLoad = () => {
```

Comment:
useCallback

Reply:
Same as prev:  loadingStory.value is useSharedValue